cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedHBaseMetricsTable.java

```diff
@@ -59,7 +109,9 @@ public CombinedHBaseMetricsTable(MetricsTable v2HBaseTable, MetricsTable v3HBase
 
   @Override
   public void put(SortedMap<byte[], ? extends SortedMap<byte[], Long>> updates) {
-    deleteColumns(updates);
+    if (tableExists.apply(resolution)) {
+      deleteColumns(updates);
```

Comment:
you check whether the table exists before every single put? How is that going to perform?